src/language-service/src/definition/includes.ts

```diff
@@ -16,7 +16,7 @@ export class IncludeDefinitionProvider implements DefinitionProvider {
       return [];
     }
     const includeType = matches[2];
-    const whatToInclude = `${matches[5]}`.trim();
+    const whatToInclude = `${matches[5]}`.trim().replace(/['"]+/g, "");
```

Comment:
We should fix this in the regular expression on line 14

Reply:
Yeah that makes sense, will take a look!

Reply:
@frenck I don't think this is possible to do *inside* the regex (because matching the the include isn't the problem) the problem is after it's matched the string needs quotes replacing from it for the path to be valid. We could chain the replace onto that line though?

Reply:
Apologies, I guess I'm probably just misunderstanding something but that regex doesn't seem to work and it also doesn't seem to solve the problem whereby `getRelativePathAsFileUri` needs to be passed a filename that doesn't contain quotes on line `23`.

Surely the job of the regex is to *match* a valid filepath, not to trim excess quotes etc?

Reply:
I think I must be misunderstanding something here @frenck so I apologise but I don't get how optimising the regex solves the problem where on line `23` we pass a file path to `getRelativePathAsFileUri` that *can* contain quotes. 

The regex doesn't strip the quotes does it? I believe it just matches a line that contains `!include` or a variation thereof?

Anyway, despite this, I tried implementing your regex including updating line 19 to point to index 4 in the regex array but it didn't work. The problem is this is my first time trying to debug a VSCode extension and I'm having difficulty logging anything to the console.

Reply:
It does strip off the quotes. The quotes are in capture group 3, which is reused at the end of the regular expression as a backreference (so the opening and closing quotes match). Capture group 4 doesn't contain the quotes as a result.

I've put together a small test for showing the result:

<https://regexr.com/5gfbm>

Reply:
To test this:

```js
    const matches = /(.*)(!include\S*)\s+(["'])?(.+)\3/.exec(line);
    if (!matches || matches.length !== 5) {
      return [];
    }
````

```js
    const whatToInclude = `${matches[4]}`.trim();
```

That should work.

Reply:
That is a good looking regex Frenck you proposed, let me get you a coffee...

Reply:
Clearly I know nothing about regexes! Cheers @frenck

Reply:
@frenck & @keesschollaart81 have updated the code with the new regex and it's all working fine